exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryWrapper.java

```diff
@@ -126,14 +128,23 @@ private float getHeapUsage() {
   public static class QueryResult {
     private final String queryId;
     public final Collection<String> columns;
+    public final HashMap<String, ArrayList> metadata;
     public final List<Map<String, String>> rows;
 
     public QueryResult(QueryId queryId, Collection<String> columns, List<Map<String, String>> rows) {
       this.queryId = QueryIdHelper.getQueryId(queryId);
       this.columns = columns;
+      this.metadata = new HashMap<String,ArrayList>();
       this.rows = rows;
     }
 
+    public QueryResult(QueryId queryId, WebUserConnection webUserConnection, List<Map<String, String>> rows) {
```

Comment:
Why we have two constructors now?

Reply:
I left the original constructor because I wasn't sure if it was used elsewhere and I didn't want to break anything.  The new one passes the `webUserConnection` directly to the `QueryResult` and enables you to access all the newly added properties.

I can remove the old one if you'd prefer.

Reply:
Please make sure it's not and remove if needed.

Reply:
Fixed